bpdm-pool/src/main/kotlin/org/eclipse/tractusx/bpdm/pool/repository/LogisticAddressRepository.kt

```
@@ -95,4 +98,60 @@ interface LogisticAddressRepository : JpaRepository<LogisticAddressDb, Long>, Jp
+                AND (
+                    CAST(:#{#searchRequest.street?.name} AS text) IS NULL
+                    OR bpdm.normalize_name(la.phy_street_name) LIKE '%' || bpdm.normalize_name(:#{#searchRequest.street?.name}) || '%'
+                    OR la.phy_street_name ~ ('.*' || REPLACE(:#{#searchRequest.street?.name}, '_', '.') || '.*')
+                )
+                AND ( 
+                    CAST(:#{#searchRequest.postalCode} AS text) IS NULL 
+                    OR la.phy_postcode = :#{#searchRequest.postalCode} 
+                )
+                AND (
+                    CAST(:#{#searchRequest.city} AS text) IS NULL
+                    OR bpdm.normalize_name(la.phy_city) LIKE '%' || bpdm.normalize_name(:#{#searchRequest.city}) || '%'
+                    OR la.phy_city ~ ('.*' || REPLACE(:#{#searchRequest.city}, '_', '.') || '.*')
+                )
+                AND (
+                    CAST(:#{#searchRequest.country} AS text) IS NULL
+                    OR la.phy_country = :#{#searchRequest.country}
+                )
+                AND (
+                       (
+                            CAST(:#{#isLegalEntity} AS boolean) = true
+                            AND le.legal_address_id = la.id
+                       )
+                    OR (
+                            CAST(:#{#isSite} AS boolean) = true
+                            AND s.main_address_id = la.id
+                       )
+                    OR (
+                            CAST(:#{#isAdditionalAddress} AS boolean) = true
+                            AND le.legal_address_id != la.id
+                            AND (s.id IS NULL OR s.main_address_id != la.id)
+                       )
+                )             
+            """,
+            nativeQuery = true
+    )
+    fun searchBusinessPartner(searchRequest: LegalEntityPropertiesSearchRequest, isLegalEntity: Boolean, isSite: Boolean, isAdditionalAddress: Boolean): Set<LogisticAddressDb>
```

General notes for that query:

At the moment the query is error prone and will fail silently if the search request changes.

Also a normalizations are being executed on the fly for both columns and parameters. This is convenient but could lead to performance issues. We should keep a close eye on the performance for this feature and introduce optimizations if needed in the future.

Also, I recommend to convert this huge query to a JPA specifications to increase the maintainability of this database query.